src/System.Net.HttpListener/src/System/Net/Managed/HttpRequestStream.Managed.cs

```diff
@@ -138,7 +154,7 @@ protected virtual IAsyncResult BeginReadCore(byte[] buffer, int offset, int size
             // for HTTP pipelining
             if (_remainingBody >= 0 && size > _remainingBody)
             {
-                size = (int)Math.Min(int.MaxValue, _remainingBody);
+                size = (int)Math.Min(_remainingBody, (long)size);
```

Comment:
What happens if size end up being 0? Do we potentially hang in a 0-byte read if the connection is left open?